python/ray/data/tests/test_consumption.py

```diff
@@ -437,11 +444,97 @@ def my_dummy_fn(x):
 
     ds = ray.data.range(10, override_num_blocks=10)
     ds = ds.map_batches(my_dummy_fn)
-    assert repr(ds) == (
+    assert ds._plan.get_plan_as_string(type(ds)) == (
         "MapBatches(my_dummy_fn)\n+- Dataset(num_rows=10, schema={id: int64})"
     )
 
 
+def test_dataset_repr(ray_start_regular_shared, restore_data_context):
+    ds = ray.data.range(5)
+    text = repr(ds)
+    assert "Dataset isn't materialized" in text
+    assert "shape: (5, 1)" in text
+    assert "│ id" in text
```

Comment:
Also, should we explicitly assert that the `repr` looks like the expected string? I think it might be easier to read and catch more errors, but I guess the tradeoff is that it'd make the test pretty brittle